dolphin/api/v1/storages.py

```diff
@@ -11,17 +11,42 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import copy
 
 from six.moves import http_client
 import webob
 
+from dolphin import db
 from dolphin.api.common import wsgi
 
 
+def build_storages(storages):
+    views = [build_storage(storage)
+             for storage in storages]
+    return dict(storage=views)
+
+
+def build_storage(storage):
+    view = copy.deepcopy(storage)
+    return view
+
 class StorageController(wsgi.Controller):
 
     def index(self, req):
-        return dict(name="Storage 1")
+        storage_all = db.storage_get_all()
+        search_opts = [
+            'name',
+            'vendor',
+            'model',
+            'status',
+        ]
+        for search_opt in search_opts:
+            if search_opt in req.GET:
```

Comment:
is it better to use db filter capability?

Reply:
That will make the DB get function complicated, data filtering can be done here,

Reply:
moved to DB layer